client/my-sites/email/titan-new-mailbox-list/index.jsx

```diff
@@ -75,29 +80,40 @@ const TitanNewMailboxList = ( {
 	return (
 		<div className="titan-new-mailbox-list__main">
 			{ mailboxes.map( ( mailbox, index ) => (
-				<TitanNewMailbox
-					key={ mailbox.uuid }
-					onMailboxAdd={ onMailboxAdd }
-					onMailboxRemove={ onMailboxRemove( mailboxes, mailbox.uuid ) }
-					onMailboxValueChange={ onMailboxValueChange( mailbox.uuid ) }
-					mailbox={ mailbox }
-					onReturnKeyPress={ onReturnKeyPress }
-					showAllErrors={ validatedMailboxUuids.includes( mailbox.uuid ) }
-					showLabels={ showLabels }
-					showTrashButton={ index > 0 }
-				/>
+				<React.Fragment key={ `${ mailbox.uuid }:${ index }` }>
+					{ index > 0 && (
+						<CardHeading tagName="h3" size={ 20 }>
+							Mailbox { String( index + 1 ).padStart( 2, '0' ) }
+						</CardHeading>
+					) }
+					<TitanNewMailbox
+						onMailboxValueChange={ onMailboxValueChange( mailbox.uuid ) }
+						mailbox={ mailbox }
+						onReturnKeyPress={ onReturnKeyPress }
+						showAllErrors={ validatedMailboxUuids.includes( mailbox.uuid ) }
+						showLabels={ showLabels }
+					/>
+
+					<div className="titan-new-mailbox-list__actions">
+						{ showAddAnotherMailboxButton && (
+							<Button onClick={ () => onMailboxAdd( index ) }>
+								<Gridicon icon="plus" />
+								<span>{ translate( 'Add another mailbox' ) }</span>
+							</Button>
+						) }
+						{ index > 0 && (
+							<Button onClick={ onMailboxRemove( mailboxes, mailbox.uuid ) }>
+								<Gridicon icon="trash" />
+								<span>{ translate( 'Remove this mailbox' ) }</span>
+							</Button>
+						) }
+					</div>
+
+					<hr className="titan-new-mailbox-list__separator" />
```

Comment:
Can we push these additional elements and UX to a separate PR please? This will affect all locations where we're adding Titan mailboxes, and makes the UX inconsistent with adding G Suite/Google Workspace mailboxes.

I think we'll want to look at this in terms of whether it's a generally useful UX improvement, so I don't want this PR to get held up by that discussion.

Reply:
Good call and I'm sorry about making the wrong assumption here too (that this change will be carried over to other places). Will work to fork out a separate PR for this.

Reply:
@daledupreez have moved these changes to #55529